src/main/resources/static/js/results.js

```diff
@@ -1,50 +1,96 @@
+/**
+ * Based on the selected values of Course, Category or Learning Outcomes
+ * a CSV files is generated with that data
+ */
 function exportToCSV() {
 
-    var url = "/export/";
-
-    //get the selected values to export CSV
-    var selectedVals = $("div.card-body input:checked").next('label').map(function () {
+    //Get the selected values to export CSV
+    var selectedVals = $("#cvsExportVals input:checked").next('label').map(function () {
         return $(this).text();
     }).get();
 
-    //checking if any values were selected
+    var colIndex = [];
+
+    //Checking if any values were selected
     if (selectedVals.length !== 0) {
-        var csvExportValsArr = [];
 
-        //populate the array as per the selected values
+        //Populate the array as per the selected values
         if (selectedVals.includes("Course")) {
-            csvExportValsArr.push("Course: ");
-            populateCsvVals(csvExportValsArr, "td:eq(0)");
-            csvExportValsArr.push("$");
+            colIndex.push(0);
+
         }
         if (selectedVals.includes("Category")) {
-            csvExportValsArr.push("Category: ");
-            populateCsvVals(csvExportValsArr, "td:eq(2)");
-            csvExportValsArr.push("$");
+            colIndex.push(2);
         }
         if (selectedVals.includes("Learning Outcomes")) {
-            csvExportValsArr.push("Learning Outcomes: ");
-            populateCsvVals(csvExportValsArr, "td:eq(3)");
+            colIndex.push(3);
         }
 
-        var joinedVals = csvExportValsArr.join(" , ");
-        url += joinedVals;
+        var csvtitle = populateCsvData(colIndex, 'th');
+        var csvContents = populateCsvData(colIndex, 'td');
+        var csvData = csvtitle + '\r\n' + csvContents;
+
+        downloadCsvFile(csvData);
+
     } else {
-        url += " ";
+        alert("Select values to export");
     }
 
-    //replaces the specific fragment
-    $("#exportCsv").load(encodeURI(url));
+}
+
```

Comment:
I found this code (here to below) on stackoverflow and just tweaked it to what we need